src/functions/advanced-functions/event.php

```diff
@@ -24,78 +26,79 @@
-	 *     @type string $FeaturedImage           URL or ID of a featured image.
-	 *     @type string $Venue                   Array of data to create or update an Venue to be associated with the Event {@link tribe_create_venue}.
-	 *     @type string $Organizer               Array of data to create or update an Organizer to be associated with the Event {@link tribe_create_organizer}.
-	 *     @type string $_ecp_custom_[ID]        Pro Custom fields (Events Calendar Pro only).
-	 * }
+	 * @type array  $tags_input                  Array of tag names, slugs, or IDs. Default empty.
+	 * @type array  $tax_input                   Array of taxonomy terms keyed by their taxonomy name. Default empty.
+	 * @type array  $meta_input                  Array of post meta values keyed by their post meta key. Default empty.
+	 * @type string $EventStartDate              Start date of event (required).
+	 * @type string $EventEndDate                End date of event (required).
+	 * @type bool   $EventAllDay                 Set to true if event has no start / end time and should run all day.
+	 * @type string $EventStartHour              Event start hour (01-12 if `EventStartMeridian` is also passed, else 00-23).
+	 * @type string $EventStartMinute            Event start minute (00-59).
+	 * @type string $EventStartMeridian          Event start meridian (am or pm).
+	 * @type string $EventEndHour                Event end hour (01-12 if `EventEndMeridian` is also passed, else 00-23).
+	 * @type string $EventEndMinute              Event end minute (00-59).
+	 * @type string $EventEndMeridian            Event end meridian (am or pm).
+	 * @type bool   $EventHideFromUpcoming       Set to true to hide this Event from the upcoming list view.
+	 * @type bool   $EventShowMapLink            Set to true to display a link to the map in the Event view.
+	 * @type string $EventShowMap                Set to true to embed the map in the Event view.
+	 * @type string $EventCost                   Default cost of the Event.
+	 * @type string $EventURL                    Link to the Event Website or Third-Party page.
+	 * @type string $FeaturedImage               URL or ID of a featured image.
+	 * @type string $Venue                       Array of data to create or update an Venue to be associated with the Event {@link tribe_create_venue}.
+	 * @type string $Organizer                   Array of data to create or update an Organizer to be associated with the Event {@link tribe_create_organizer}.
+	 * @type string $_ecp_custom_                [ID]        Pro Custom fields (Events Calendar Pro only).
+	 *                                           }
 	 *
 	 * @return int|bool ID of the event that was created. False if insert failed.
+	 * @category Events
+	 *
+	 * @version  TBD
+	 * @since    TBD Updated docblock to indicate future deprecation of function
 	 */
-	function tribe_create_event( $args ) {
+	function tribe_create_event( array $args ): bool|int {
```

Comment:
We have to be very careful about changes of this nature, for example:
1. Multiple expected return types are not available on our base PHP version 7.4
2. We should avoid adding type hinting on existing method that have public usage, since WP development hasn't fully embraced these yet we can cause Fatals that would be on our software and we would be the ones responsible for the support in some way.